pkg/master/master.go

```diff
@@ -242,6 +242,10 @@ type Master struct {
 
 	// storage for third party objects
 	thirdPartyStorage storage.Interface
+	// map from api path to storage for those objects
+	thirdPartyResources map[string]*thirdpartyresourcedataetcd.REST
+	// protects the map
+	thirdPartyResourcesLock sync.RWMutex
```

Comment:
If we did that (and I'm open to it, I suppose) you would still need to expose functions for mucking around with the go-restful web services that would look pretty similar to these functions.  I guess the other object could muck with `handlerContainer` directly, but that seems to break encapsulation boundaries.